cmd/lakefs/cmd/root.go

```diff
@@ -50,42 +52,58 @@ func init() {
 func initConfig() {
 	logger := logging.Default().WithField("phase", "startup")
 	if cfgFile != "" {
-		logger.WithField("file", cfgFile).Info("configuration file")
+		logger.WithField("file", cfgFile).Info("Configuration file")
 		// Use config file from the flag.
 		viper.SetConfigFile(cfgFile)
 	} else {
-		logger.Info("search for configuration file .lakefs")
-		// Find home directory.
-		home, err := homedir.Dir()
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		}
-
-		// Search config in home directory with name ".lakefs" (without extension).
-		viper.AddConfigPath(home)
 		viper.SetConfigType("yaml")
-		viper.SetConfigName(".lakefs")
+		viper.SetConfigName("config")
+		viper.AddConfigPath("/etc/lakefs")
+		viper.AddConfigPath(path.Join(getHomeDir(), ".lakefs"))
+		viper.AddConfigPath(".")
 	}
 
 	viper.SetEnvPrefix("LAKEFS")
 	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_")) // support nested config
 	// read in environment variables
 	viper.AutomaticEnv()
 
-	// If a config file is found, read it in.
+	// read configuration file
 	err := viper.ReadInConfig()
-	logger = logger.WithField("file", viper.ConfigFileUsed())
-
-	if err == nil {
-		logger.Info("loaded configuration from file")
-	} else if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
-		logger.WithError(err).Fatal("failed to read config file")
+	logger = logger.WithField("file", viper.ConfigFileUsed()) // should be called after SetConfigFile
+	var errFileNotFound viper.ConfigFileNotFoundError
+	if err != nil && !errors.As(err, &errFileNotFound) {
+		logger.WithError(err).Fatal("Failed to find a config file")
+	}
+	// fallback - try to load the previous supported $HOME/.lakefs.yaml
+	//   if err is set it will be file-not-found based on previous check
```

Comment:
I don't understand this.  Didn't line 62 _already_ cause Viper to search for this file?

Reply:
The package supports an additional search path - this case is to support an additional filename.
The current search lookup for `.lakefs.yaml` (starts with dot) - in order to support it, we try to find a configuration find (the new standard) first and need to try to read the specific one - unless the user specified it already.